task8/ProfitApp/src/main/java/com/bsu/fpmi/ProfitApp/Ad.java

```diff
@@ -0,0 +1,159 @@
+package com.bsu.fpmi.ProfitApp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.List;
+
+public class Ad {
+
+    private String id;
+    private String description;
+    private String createdAt;
+    private String link;
+    private String vendor;
+    private String photoLink;
+    private List<String> hashTags;
+    private String discount;
+    private String validUntil;
+    private Double rating;
+    private List<String>reviews;
+
+    private boolean isDateCorrect(String date) {
```

Comment:
Please observe the normative class composition order: fields, then constructors, then instance methods, then static methods. Methods themselves are ordered: public, then package-private, then private.